### PR TITLE
Added checkbox support

### LINKED ISSUE #12: Update bootstrap-input to allow checkbox
Example:
{{bootstrap-input
    type="checkbox"
    label='Unlimited'
    value=product.unlimited}}

### PR DESCRIPTION
**Pull request by jweakley**

### CODE REVIEW

**Comment by jweakley** [comment]
I'll add the version bump to this PR once it has been reviewed/approved.

File: addon/templates/components/bootstrap-input.hbs [hunk 1]
```diff
@@ -4,6 +4,32 @@
   <label for="{{inputId}}" class="control-label {{if hasColumns labelColumnCss ''}} {{if customLabelCss customLabelCss ''}} {{if srOnly 'sr-only' ''}}">{{label}}</label>
   {{#if hasColumns}}
     <div class="{{inputColumnCss}}">
+      {{#if checkboxType}}
```
Comment (Samsinite) [inline addon/templates/components/bootstrap-input.hbs]: No need for conditional, always pass checked=checked. If it isn't null, input will spit out an error. If you want an additional check in the component, decorate the checked property and use `Ember.assert`.

File: addon/templates/components/bootstrap-input.hbs [hunk 2]
```diff
@@ -4,6 +4,32 @@
...
+  {{else}}
+    {{#if checkboxType}}
+      {{input checked=checked
+          elementId=inputId
+          type=type
+          placeholder=placeholder
+          readonly=readonly
+          disabled=disabled}}
+    {{else}}
       {{input value=value
```
Comment (Samsinite) [inline addon/templates/components/bootstrap-input.hbs]: checked=checked also needs set here.

**Comment by jweakley** [comment]
@Samsinite, I altered this to allow for the following syntax:
`{{bootstrap-input type="checkbox" label='Unlimited' value=product.unlimited}}` instead of `{{bootstrap-input type="checkbox" label='Unlimited' checked=product.unlimited}}`

I thought it might be one less thing to remember if you always pass the value in as value and let the addon handle how it works, rather than have the consumer of this addon need to remember that in the use case of checkbox you don't use `value`, you use `checked`.

**Comment by Samsinite** [comment]
I like the single interface idea, but lets keep the template as simple as possible and related to markup only. You can decorate the value and checked properties to achieve this:
```js
valueDecorator: Ember.computed('value', 'type', function() { ... },
checkedDecorator: Ember.computed('value', 'type', function() { ... }
```
```html
{{input value=valueDecorator checked=checkedDecorator ...}}
```

**Comment by Samsinite** [comment]
Also, have you eaten your own dog food on this (npm linked and tested it in an app)? If not, please at least manually test it in the dummy app inside of the addon or write some [component integration tests](http://alisdair.mcdiarmid.org/ember-component-integration-tests/) to make sure the checkboxes are working correctly and that nothing else is broken.

**Comment by jweakley** [comment]
Updated and manually tested.

**Pull request merged**